simulation/chart.py

```diff
@@ -1,14 +1,134 @@
 # Copyright 2022-, Semiotic AI, Inc.
 # SPDX-License-Identifier: Apache-2.0
 
+from enum import Enum
 from subprocess import Popen
-from typing import Dict, Optional, Tuple
+from typing import Any, Dict, NamedTuple, Optional, Tuple, TypedDict
 
 import ffmpeg
 import pyqtgraph as pg
 from pyqtgraph.Qt import QtCore, QtGui
 
 
+class RGBAColor(NamedTuple):
+    """An RGB color value represents RED, GREEN, and BLUE light sources
+    with alpha channel- which specifies the opacity for a color."""
+
+    R: int
+    """Red [0-255]"""
+    G: int
+    """Green [0-255]"""
+    B: int
+    """Blue [0-255]"""
+    A: Optional[int]
+    """Alpha [0-255]"""
+
+
+class NamedColor(Enum):
+    "Single-character string representing a predefined color"
+    r = "r"
+    """Red"""
+    g = "g"
+    """Green"""
+    b = "b"
+    """Blue"""
+    c = "c"
+    """Cyan"""
+    m = "m"
+    """Magenta"""
+    y = "ky"
+    """Yellow"""
+    k = "k"
+    """Key (black)"""
+    w = "w"
+    """White"""
+
+
+class IndexedColor(NamedTuple):
+    """Color from a single index. Useful for stepping through a predefined list of colors."""
+
+    index: int
+    hues: int
+
+
+Color = RGBAColor | NamedColor | IndexedColor | int | float | str
+"""Color based on model, predefined values, single greyscale value (0.0 - 1.0) , color index or string representing color (“#RGB”,“#RGBA”,“#RRGGBB”,“#RRGGBBAA”)."""
+
+
+def _make_color(color: Color) -> Any:
```

Comment:
This would be cleaner: https://coady.github.io/multimethod/

Reply:
https://github.com/semiotic-ai/autoagora-agents/pull/21/commits/a65c7aec3b369afd02036d66bdd5b9834834dd44